src/OrchardCore.Modules/OrchardCore.Email/Drivers/SmtpSettingsDisplayDriver.cs

```diff
@@ -83,28 +99,106 @@ public override async Task<IDisplayResult> UpdateAsync(SmtpSettings section, Bui
                 return null;
             }
 
-            if (context.GroupId.Equals(GroupId, StringComparison.OrdinalIgnoreCase))
+            if (!context.GroupId.Equals(EmailSettings.GroupId, StringComparison.OrdinalIgnoreCase))
+            {
+                return null;
+            }
+
+            var model = new SmtpSettingsViewModel();
+
+            if (await context.Updater.TryUpdateModelAsync(model, Prefix))
             {
-                var previousPassword = section.Password;
-                await context.Updater.TryUpdateModelAsync(section, Prefix);
+                var emailSettings = site.As<EmailSettings>();
+
+                var hasChanges = model.IsEnabled != settings.IsEnabled;
 
-                // Restore password if the input is empty, meaning that it has not been reset.
-                if (string.IsNullOrWhiteSpace(section.Password))
+                if (!model.IsEnabled)
                 {
-                    section.Password = previousPassword;
+                    if (hasChanges && emailSettings.DefaultProviderName == SmtpEmailProvider.TechnicalName)
+                    {
+                        emailSettings.DefaultProviderName = null;
+
+                        site.Put(emailSettings);
+
+                        await _notifier.WarningAsync(H["You have successfully disabled the default Email provider. The Email service is now disable and will remain disabled until you designate a new default provider."]);
+                    }
+
+                    settings.IsEnabled = false;
                 }
                 else
                 {
-                    // encrypt the password
-                    var protector = _dataProtectionProvider.CreateProtector(nameof(SmtpSettingsConfiguration));
-                    section.Password = protector.Protect(section.Password);
+                    if (string.IsNullOrEmpty(emailSettings.DefaultProviderName))
+                    {
+                        await _notifier.WarningAsync(H["No designated default email provider is currently enabled. Please select and set one of the available email providers as the default."]);
```

Comment:
Actually, we don't need the warning. If the user does not specify which provider to use, by default the email provider will select the last enabled provider if any otherwise, it would use the last registered provider.